types/chai/index.d.ts

```diff
@@ -422,110 +422,101 @@ declare namespace Chai {
         /**
          * Throws a failure.
          *
-         * @type T   Type of the objects.
          * @param actual   Actual value.
          * @param expected   Potential expected value.
          * @param message    Message to display on error.
          * @param operator   Comparison operator, if not strict equality.
          * @remarks Node.js assert module-compatible.
          */
-        fail<T>(actual: T, expected: T, message?: string, operator?: Operator): never;
+        fail(actual: unknown, expected: unknown, message?: string, operator?: Operator): never;
 
         /**
          * Asserts that object is truthy.
          *
-         * @type T   Type of object.
          * @param object   Object to test.
          * @param message    Message to display on error.
          */
-        isOk<T>(value: T, message?: string): void;
+        isOk(value: unknown, message?: string): asserts value;
 
         /**
          * Asserts that object is truthy.
          *
-         * @type T   Type of object.
          * @param object   Object to test.
          * @param message    Message to display on error.
          */
-        ok<T>(value: T, message?: string): void;
+        ok(value: unknown, message?: string): asserts value;
 
         /**
          * Asserts that object is falsy.
          *
-         * @type T   Type of object.
          * @param object   Object to test.
          * @param message    Message to display on error.
          */
-        isNotOk<T>(value: T, message?: string): void;
+        isNotOk(value: object, message?: string): asserts value is never;
+        isNotOk(value: unknown, message?: string): asserts value is "" | 0 | false | null | undefined;
 
         /**
          * Asserts that object is falsy.
          *
-         * @type T   Type of object.
          * @param object   Object to test.
          * @param message    Message to display on error.
          */
-        notOk<T>(value: T, message?: string): void;
+        notOk(value: object, message?: string): asserts value is never;
+        notOk(value: unknown, message?: string): asserts value is "" | 0 | false | null | undefined;
```

Comment:
Is this a complete list of falsy values?  What about `NaN`?

Reply:
I don't think there's a literal type for `NaN` and I'm not sure how else you would express it.